maven-resolver-provider/src/main/java/org/apache/maven/repository/internal/MavenRepositorySystemUtils.java

```diff
@@ -59,8 +62,37 @@ private MavenRepositorySystemUtils() {
      * the session with authentication, mirror, proxy and other information required for your environment.
      *
      * @return The new repository system session, never {@code null}.
+     * @deprecated This method is deprecated.
      */
+    @Deprecated
     public static DefaultRepositorySystemSession newSession() {
+        DefaultArtifactTypeRegistry stereotypes = new DefaultArtifactTypeRegistry();
+        stereotypes.add(new DefaultArtifactType("pom"));
```

Comment:
Should we use `DefaultTypeRegistry` and the various providers here instead ?

Reply:
This method is deprecated and meant NOT to be used in Maven, only outside (like MIMA). Hence, there is no Maven API present either, "only" resolver stuff, so this is good as is IMHO.

Reply:
Deprecation is probably wrong.... is "deprecated" from Maven POV as it wants to inject own TypeRegistry, but again, code integrating Maven Resolver (like MIMA) does not have new Maven API on classpath, and it will want to stick to "pure" resolver classes...